File: old_code/src/utils/LexerOptions.ts
/**
 * Enum representing the token splitter type.
 */
export enum TokenSplitterType {
  None,
  Space,
  Regex
}

/**
 * Options for the lexer.
 */
export class LexerOptions {
  private returnLexerPathTokens: boolean;
  private returnIndentTokens: boolean;

  /**
   * Creates a new LexerOptions instance.
   * @param returnLexerPathTokens Whether to return lexer path tokens
   * @param returnIndentTokens Whether to return indent tokens
   */
  constructor(returnLexerPathTokens: boolean = false, returnIndentTokens: boolean = false) {
    this.returnLexerPathTokens = returnLexerPathTokens;
    this.returnIndentTokens = returnIndentTokens;
  }

  /**
   * Gets whether to return lexer path tokens.
   */
  public getReturnLexerPathTokens(): boolean {
    return this.returnLexerPathTokens;
  }

  /**
   * Sets whether to return lexer path tokens.
   * @param value The new value
   */
  public setReturnLexerPathTokens(value: boolean): void {
    this.returnLexerPathTokens = value;
  }

  /**
   * Gets whether to return indent tokens.
   */
  public getReturnIndentTokens(): boolean {
    return this.returnIndentTokens;
  }

  /**
   * Sets whether to return indent tokens.
   * @param value The new value
   */
  public setReturnIndentTokens(value: boolean): void {
    this.returnIndentTokens = value;
  }
}

File: src/StepParser/Lexer/LexerOptions.cs
namespace StepParser.Lexer;

/// <summary>
/// Enum representing token splitter types.
/// </summary>
public enum TokenSplitterType
{
    None = 0,
    Whitespace = 1,
    Regex = 2,
    Custom = 3
}

/// <summary>
/// Options for configuring the lexer.
/// </summary>
public class LexerOptions
{
    /// <summary>
    /// Gets or sets a value indicating whether to return lexer path tokens.
    /// </summary>
    public bool ReturnLexerPathTokens { get; set; } = false;

    /// <summary>
    /// Gets or sets the token splitter type.
    /// </summary>
    public TokenSplitterType TokenSplitterType { get; set; } = TokenSplitterType.None;

    /// <summary>
    /// Gets or sets the regex pattern for token splitting.
    /// </summary>
    public string? RegexTokenSplitter { get; set; }

    /// <summary>
    /// Gets or sets a value indicating whether to ignore case.
    /// </summary>
    public bool IgnoreCase { get; set; } = false;

    /// <summary>
    /// Gets or sets a value indicating whether to enable debugging.
    /// </summary>
    public bool EnableDebugging { get; set; } = false;

    /// <summary>
    /// Gets or sets the maximum number of lexer paths.
    /// </summary>
    public int MaxLexerPaths { get; set; } = 1000;
}